src/main/java/dev/thedocruby/resounding/Utils.java

```diff
@@ -129,12 +131,15 @@ public static <T> HashMap<String, T> recall(String path, Type token, Function<Li
         return output;
     }
 
-    public static <T> HashMap<String, T> resource(ResourcePack pack, String path, Type token, BiFunction<String, LinkedTreeMap, HashMap<String, T>> deserializer) {
+    public static <T> HashMap<String, T> resource(ResourcePack pack, String[] path, Type token, BiFunction<String, LinkedTreeMap, HashMap<String, T>> deserializer) {
         HashMap<String, T> output = new HashMap<>();
         InputStream input;
         // if not available, move on
-        try { input = pack.openRoot(path); }
-        catch (IOException e) { return output; }
+        try {
+            InputSupplier<InputStream> supplier = pack.openRoot(path);
+            if (supplier == null) return output;
```

Comment:
What about a `catch (NullPointerException e)` instead here?

Reply:
That is also perfectly valid. I'll change it now